MatrixSDK/Crypto/Data/Store/MXRealmCryptoStore/MXRealmCryptoStore.m

```diff
@@ -1105,6 +1128,92 @@ - (void)removeOutboundGroupSessionWithRoomId:(NSString*)roomId
     }];
 }
 
+- (void)storeSharedDevices:(MXUsersDevicesMap<NSNumber *> *)devices messageIndex:(NSUInteger) messageIndex forOutboundGroupSessionInRoomWithId:(NSString *)roomId sessionId:(NSString *)sessionId
+{
+    NSDate *startDate = [NSDate date];
+
+    RLMRealm *realm = self.realm;
+
+    [realm transactionWithBlock:^{
+        
+        for (NSString *userId in [devices userIds])
+        {
+            for (NSString *deviceId in [devices deviceIdsForUser:userId])
+            {
+                MXRealmUser *realmUser = [MXRealmUser objectsInRealm:realm where:@"userId = %@", userId].firstObject;
+                if (!realmUser)
+                {
+                    NSLog(@"[MXRealmCryptoStore] storeSharedDevices cannot find user with the ID %@", userId);
+                    continue;
+                }
+
+                MXRealmDeviceInfo *realmDevice = [[realmUser.devices objectsWhere:@"deviceId = %@", deviceId] firstObject];
+                if (!realmDevice)
+                {
+                    NSLog(@"[MXRealmCryptoStore] storeSharedDevices cannot find device with the ID %@", deviceId);
+                    continue;
+                }
+
+                MXRealmSharedOutboundSession *sharedInfo = [[MXRealmSharedOutboundSession alloc] initWithValue: @{
+                    @"uniqueKey": [MXRealmSharedOutboundSession generateUniqueKey],
+                    @"roomId": roomId,
+                    @"sessionId": sessionId,
+                    @"device": realmDevice,
+                    @"messageIndex": @(messageIndex)
+                }];
+                [realm addObject:sharedInfo];
+            }
+        }
+    }];
+
+    NSLog(@"[MXRealmCryptoStore] storeSharedDevices (count: %tu) in %.3fms", devices.count, [[NSDate date] timeIntervalSinceDate:startDate] * 1000);
+}
+
+- (MXUsersDevicesMap<NSNumber *> *)sharedDevicesForOutboundGroupSessionInRoomWithId:(NSString *)roomId sessionId:(NSString *)sessionId
+{
+    MXUsersDevicesMap<NSNumber *> *devices = [MXUsersDevicesMap new];
+
+    RLMRealm *realm = self.realm;
+    
+    [realm transactionWithBlock:^{
```

Comment:
We do not need `transactionWithBlock` for read operations.

Reply:
`transactionWithBlock` removed